clients/java/signalr/src/main/java/JsonHubProtocol.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.util.ArrayList;
+
+public class JsonHubProtocol implements IHubProtocol{
+    private JsonParser jsonParser = new JsonParser();
+    private static final String RECORD_SEPARATOR = "\u001e";
+
+    @Override
+    public String name() {
+        return "json";
+    }
+
+    @Override
+    public int version() {
+        return 1;
+    }
+
+    @Override
+    public TransferFormat transferFormat() {
+        return TransferFormat.Text;
+    }
+
+    @Override
+    public InvocationMessage[] parseMessages(String payload) {
```

Comment:
rant: I _love_ how Java insists on a curly bracket at the end and then you insert a new line  to make code readable.